library/std/src/env.rs

```diff
@@ -318,10 +318,52 @@ impl Error for VarError {
 ///
 /// # Safety
 ///
-/// Even though this function is currently not marked as `unsafe`, it needs to
-/// be because invoking it can cause undefined behaviour. The function will be
-/// marked `unsafe` in a future version of Rust. This is tracked in
-/// [rust#27970](https://github.com/rust-lang/rust/issues/27970).
+/// This function is safe to call in a single-threaded program.
+///
+/// In multi-threaded programs, you must ensure that are no other threads
+/// concurrently writing or *reading*(!) from the environment through functions
+/// other than the ones in this module. You are responsible for figuring out
+/// how to achieve this, but we strongly suggest not using `set_var` or
+/// `remove_var` in multi-threaded programs at all.
```

Comment:
Would it also makes sense to mention that this is always safe to call on Windows (and maybe listing other platforms that are well behaved?). My comment is inspired by https://github.com/rust-lang/rust/issues/124866#issuecomment-2102404431 but instead of adding a safe API for Windows, we could at least document it as such.

Reply:
Done.